mbtools/models.py

```diff
@@ -168,14 +168,15 @@ def __init__(self, parent, location):
         self.parent = parent
         self.location = location
         self.etree_fragments = []
+        self.unnested_content = []
 
         # Catch strings that exist without html tags
         temp = html.fragments_fromstring(self.parent.text)
         for fragment in temp:
             if type(fragment) != html.HtmlElement:
-                p_element = etree.Element('p')
-                p_element.text = fragment
-                self.etree_fragments.append(p_element)
+                self.unnested_content.append(fragment)
+            elif (fragment.tail is not None and fragment.tail.strip()):
+                self.unnested_content.append(fragment.tail)
```

Comment:
Just to confirm my understanding: The intent of this change is to store the unnested content in the array but essentially hide it while pushing "valid" content into `etree_fragments`, correct? If so, there's maybe a small issue here (which perhaps doesn't really impact anything in practice with our current usage) which is that on this code path you may have wanted to do:

```python
self.unnested_content.append(fragment.tail)
fragment.tail = None # Or whatever the appropriate "clear" value is
self.etree_fragments.append(fragment)
```

That way if you had something like `<p>Para</p>text` the `<p>Para</p>` still gets pushed as a fragment into the internal state.

Does :point_up: sound valid to you, or am I perhaps misunderstanding something?

Reply:
I think that may be the case, yes - the tail will get pushed with the fragment. And like you said, I don't think in our processing we'll ever really interact with the tail of the fragment so it may be a non-issue. I'll go ahead and change it regardless.

Reply:
Ah, just to clarify I think the original code wasn't pushing the fragment at all because it had a tail (e.g. `<p>para1</p>text<p>para2</p>` would end up with 1 entry in `etree_fragments`). The change to add the additional two lines in my comment would push the fragment after clearing the `tail` content.

Reply:
Ah I see your point. Yes, that makes sense.